### DISCUSSION TITLE
Could not Load file CefSharp.Core.Runtime error with CefSharp project built for mixed platforms - CefSharp version 100.0.140

### DISCUSSION THREAD

**Discussion by IntegraHarlan** [cefsharp/CefSharp#4065] [General]
Hi, I added a project with the ChromiumBrowser in our application. It is a large Winforms application which has a couple web services.
No CefSharp code is used in the web services.
The problem I am having is the most of the projects are build as anycpu.
My project is a simple form that loads a browser when a button on the application is pressed.
The startup project is built as anycpu, but set to prefer 32 bit. For reasons I cannot explain, this cannot be changed.
When I build my project with the platform target set to anycpu, it will error with this message:  "Could not load file or assembly CefSharp.Core.Runtime" when I try to load it.
If I change the platform target for my project to x86, the project form will load, but my webservices give the following error when I try to browse to them: "Could not load file or assembly 'CefSharp.BrowserSubprocess.Core.DLL"
I am not sure why this is happening. It appears that it has something to do with how CefSharp and IIS work together?

I would like to be able to build my project as anycpu and start it with out errors.
It appears that since I am loaded my form from a project that is built to prefer 32 bit, and my project is being loaded as 32 bit.
The CefSharp appears to try to load the 64 bit dll's and errors.
I tried the [this](https://github.com/cefsharp/CefSharp/issues/1714) solution, but it did not help.

Is there a way to load my project as anycpu and have load with out breaking my web services?

**Comment by amaitland** [comment]
> I am not sure why this is happening. It appears that it has something to do with how CefSharp and IIS work together?

Are you trying to integrate CefSharp into IIS? The requirements imposed upon us from chromium means CefSharp is not well suited to integrate directly into IIS. 

If you need to specify a specific architecture you can set CefSharpPlatformTargetOverride in your csproj. https://github.com/cefsharp/CefSharp/blob/cefsharp/100/NuGet/CefSharp.Common.targets#L128

Unfortunately I don't have a clear picture of how your project is structured. Need a clearer picture if you require further assistance.

**Reply by IntegraHarlan** [reply]
Thank you for your quick response Amaitland.
I am not trying to integrate CefSharp into IIS. I was just speculating why it broke my web services.

I did figure out what my problem was however. The information in [Issue #1714](https://github.com/cefsharp/CefSharp/issues/1714) was close to what I needed.
The information in that issue indicated that the app.config file would automatically be updated by following the steps in the last item of that thread. This did not happen. I manually updated the app.config file in my project, but that did not fix my issue. I had to update the app.config in the startup project also. Once both the app.config files where updated as indicated in Issue #1714, I was able to load my form when it was build as anycpu and still have my web services work

Thanks

**Reply by amaitland** [reply]
The instructions should be updated to clarify that the app.config approach will only work if you are installing CefSharp directly in an Exe/WinExe proj. I don't believe .Net supports having an app.config in a library project. At least not in the context of the assemblyBinding used to load the dlls.

If you are using AnyCpu Prefer32Bit true then you are producing a 32bit executable, the naming is misleading. So including the 32bit and 64bit binaries is unnessicary as only the 32bit binaries will be used. Setting CefSharpPlatformTargetOverride to x86 would result in a smaller application.

**Reply by IntegraHarlan** [reply]
Amaitland,
Using the 32 bit binaries only does not work for me as it breaks the web services that I have in the application. I am not sure why this happens as I am not using CefSharp in them. 
The project/form that I am using CefSharp is built as anyCpu with prefer32bit false.
The startup project that my form is loaded from is built with prefer32bit set to true.
It appears that you are correct about the .Net does not support app.config in a library project.
updating the app.config in the start up project works for me. 
I will try to set the CefSharpPlatformTargetOverride to x86 to see if that works.
Do you know if I need to do that on my project that contains the CefSharp code or the startup project?
Thanks

**Reply by amaitland** [reply]
> Using the 32 bit binaries only does not work for me as it breaks the web services that I have in the application. I am not sure why this happens as I am not using CefSharp in them.

If they are AnyCPU running on an x64 host then they can only load AnyCPU or x64 libraries. Nothing unusual here.

> The startup project that my form is loaded from is built with prefer32bit set to true.

This means your application is 32bit (x86).

> Do you know if I need to do that on my project that contains the CefSharp code or the startup project?

Set the property in the project file which you've installed the CefSharp Nuget packages.
